app/analysis/index/controller.js

```diff
@@ -143,14 +70,97 @@ export default class AnalysisController extends Controller.extend(
   @tracked _lastPage = 0;
   @tracked totalTime = moment.duration();
   @tracked totalItems = A();
-  @tracked selectedReportIds;
+  @tracked selectedReportIds = A();
   @tracked _dataCache = A();
 
   @tracked user;
   @tracked reviewer;
   @tracked customer;
   @tracked project;
   @tracked task;
+  @tracked rejected;
+  @tracked editable;
+  @tracked billingType;
+  @tracked fromDate;
+  @tracked toDate;
+  @tracked review;
+  @tracked notBillable;
+  @tracked verified;
+  @tracked billed;
+  @tracked costCenter;
+  @tracked ordering = "-date";
+
+  @action
+  changeFromDate(date) {
+    this.fromDate = serializeMoment(date);
+    this._reset();
+  }
+
+  get getFromDate() {
+    return deserializeMoment(this.fromDate);
+  }
+
+  @action
+  changeOrdering(ordering) {
+    this.ordering = `${ordering},id`;
+    this._reset();
+  }
+
+  get getOrdering() {
+    return this.ordering.replace(",id", "");
+  }
+
+  @action
+  changeToDate(date) {
+    this.toDate = serializeMoment(date);
+    this._reset();
+  }
+
+  get getToDate() {
+    return deserializeMoment(this.toDate);
+  }
+
+  @action
+  updateReview(value) {
+    this.review = value;
+    this._reset();
+  }
+
+  @action
+  updateNotBillable(value) {
+    this.notBillable = value;
+    this._reset();
+  }
+
+  @action
+  updateVerified(value) {
+    this.verified = value;
+    this._reset();
+  }
+
+  @action
+  updateBilled(value) {
+    this.billed = value;
+    this._reset();
```

Comment:
This does look cumbersome and error prone. Let's keep it `DRY`. I think what we need here is some sort of `tracked` function which is triggered whenever one of those query params change.

Reply:
or at least write some generic function like 
```js
@action
updateParam(key, value) {
  this[key] =  value;
  this._reset();
}
```